src/pow/mempool.cpp

```diff
@@ -127,19 +127,31 @@ void MemPool::PushRedemptionTx(ConstTxPtr redemption) {
     redemptionTxQueue_.Put(redemption);
 }
 
-ConstTxPtr MemPool::GetRedemptionTx(bool IsFirstReg) {
+ConstTxPtr MemPool::GetRedemptionTx() {
     READER_LOCK(mutex_)
+
     if (!redemptionTxQueue_.Empty()) {
         ConstTxPtr ptr;
         redemptionTxQueue_.Take(ptr);
-        if (IsFirstReg && !ptr->IsFirstRegistration()) {
-            return nullptr;
-        }
         return ptr;
     }
+
     return nullptr;
 }
 
+ConstTxPtr MemPool::GetFirstReg() {
+    READER_LOCK(mutex_)
+
+    ConstTxPtr ptr;
+    redemptionTxQueue_.Take(ptr);
+
+    if (!ptr->IsFirstRegistration()) {
+        ptr.reset();
```

Comment:
Do we need to put it back? Or we regard this redemption tx as an invalid one?

Reply:
We should regard it as an invalid one.